src/org/mozilla/javascript/NativeArray.java

```diff
@@ -2214,11 +2215,7 @@ public ListIterator listIterator() {
 
     @Override
     public ListIterator listIterator(final int start) {
-        long longLen = length;
-        if (longLen > Integer.MAX_VALUE) {
-            throw new IllegalStateException();
-        }
-        final int len = (int) longLen;
+        final int len = size();
```

Comment:
Looks like we have this pattern at more places, maybe we can remove the duplicated code at all places

toArray()
indexOf()
lastIndexOf()

And maybe we can add some text to the illegal state exception thrown in the size() method.

Reply:
Done: https://github.com/mozilla/rhino/pull/901/commits/e6ba6409ce0399c3ae27682fa4f4d365d975cb2f